MOM6/patches/mom_cap.F90.patch

```diff
@@ -1,6 +1,272 @@
+   type(ocean_state_type),        pointer :: ocean_state_type_ptr
+   type(ice_ocean_boundary_type), pointer :: ice_ocean_boundary_type_ptr
++#ifdef _USE_GENERIC_TRACER
++  type(coupler_2d_bc_type),      pointer :: coupler_2d_bc_type_ptr
++#endif
+ end type
+ 
+ !>  Wrapper-derived type required to associate an internal state instance
+@@ -392,6 +406,10 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
+   type (ocean_public_type),      pointer :: ocean_public => NULL()
+   type (ocean_state_type),       pointer :: ocean_state => NULL()
+   type(ice_ocean_boundary_type), pointer :: Ice_ocean_boundary => NULL()
++  type(coupler_1d_bc_type),      pointer :: gas_fields_atm => NULL()
++  type(coupler_1d_bc_type),      pointer :: gas_fields_ocn => NULL()
++  type(coupler_1d_bc_type),      pointer :: gas_fluxes => NULL()
++  type(coupler_2d_bc_type),      pointer :: atm_fields => NULL()
+   type(ocean_internalstate_wrapper)      :: ocean_internalstate
+   type(ocean_grid_type),         pointer :: ocean_grid => NULL()
+   type(directories)                      :: dirs
+@@ -423,6 +441,7 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
+   character(len=512)                     :: restartfile          ! Path/Name of restart file
+   character(len=2048)                    :: restartfiles         ! Path/Name of restart files
+                                                                  ! (same as restartfile if single restart file)
++  character(240)                         :: additional_restart_dir
+   character(len=*), parameter            :: subname='(MOM_cap:InitializeAdvertise)'
+   character(len=32)                      :: calendar
+   character(len=:), allocatable          :: rpointer_filename
+@@ -498,6 +517,8 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
+   if (chkerr(rc,__LINE__,u_FILE_u)) return
+   call MOM_infra_init(mpi_comm_mom)
+ 
++  call field_manager_init
++
+   ! determine the calendar
+   if (cesm_coupled) then
+     call NUOPC_CompAttributeGet(gcomp, name="calendar", value=cvalue, &
+@@ -629,13 +650,44 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
+ 
+   endif
+ 
++  ! Set NUOPC attribute additional_restart_dir to RESTART/ if not defined
++  additional_restart_dir = "RESTART/"
++  call NUOPC_CompAttributeGet(gcomp, name="additional_restart_dir", value=cvalue, &
++        isPresent=isPresent, isSet=isSet, rc=rc)
++  if (ChkErr(rc,__LINE__,u_FILE_u)) return
++  if (isPresent .and. isSet) then
++    additional_restart_dir = slasher(cvalue)
++  else
++    call ESMF_LogWrite('MOM_cap:additional_restart_dir unset. Defaulting to '//trim(additional_restart_dir), &
++          ESMF_LOGMSG_INFO)
++  endif
++  call NUOPC_CompAttributeSet(gcomp, name="additional_restart_dir", value=additional_restart_dir, rc=rc)
++  if (chkerr(rc,__LINE__,u_FILE_u)) return
```

Comment:
Yeah (although NUOPC only really specifies how the files are handled, MOM does the writing). The name is general because it's a general config parameter that in theory it could be used by any model component handled by NUOPC. I'll see what I can do to make this clearer. One option is just to hardcode the directory (it wouldn't be the first).

Reply:
The main problem is that NUOPC wants the restart input and output directories to be `./` for all components. The restart filenames include a timestep to prevent files from being overwritten and a pointer file keeps track of the latest restart. The name of the additional restart file needed for the generic tracer fluxes is set within each generic_tracer module (i.e. I don't want to change it) and does not include a timestamp, so writing to and reading from MOMs ` restart_output_dir` and `restart_input_dir` (`=./`), respectively, causes problems.